rhine/src/FRP/Rhine/Clock/Never.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TypeFamilies #-}
+
+-- | A clock that never ticks.
+module FRP.Rhine.Clock.Never where
+
+-- base
+import Data.Time.Clock
+
+-- rhine
+import FRP.Rhine.Clock
+import FRP.Rhine.Clock.Proxy
+
+import Control.Concurrent (threadDelay)
+import Control.Monad (forever)
+import Data.Kind (Type)
+
+{- |
+A clock that never ticks.
+-}
+data Never (tag :: Type) = Never
+
+instance Clock IO (Never tag) where
+  type Time (Never _) = UTCTime
+  type Tag (Never tag) = tag
+
+  initClock _ = do
+    initialTime <- getCurrentTime
+    return
+      ( constM getCurrentTime
+          &&& arrM (const . forever . threadDelay $ 10 ^ 9)
```

Comment:
You should be able to use `constM` here.

Reply:
Excellent idea; obviously if `forever` can produce `tag`, it can also produce `(time, tag)`. :slightly_smiling_face: